### ISSUE TITLE
Flag for Disabling Admin Web GUI

### ISSUE THREAD

**Issue by labeldevops** [boringproxy/boringproxy#172]
I would rather like to disable the admin web gui - I do not need that 24/7 and no matter how secure your code might be I would like to reduce the potential attack surface by only running the web gui when needed. I did not find any option to disable the gui, so it woud be needed to implement that.

**Comment by anderspitman** [comment]
Hey @labeldevops. This is similar to #103. Would providing an option to control what address the server binds to be sufficient? That way you can bind it to localhost and no remote processes will be able to access it.

**Comment by WGrobler** [comment]
@anderspitman, I agree that controlling the port that the server binds to will solve this problem and #103 . It might not be necessary to bind it only to 127.0.0.1, it can still bind to 0.0.0.0, if someone wants to limit access from outside they can block external access to the web UI port using the firewall. Or they can create firewall rules to only allow access to that port from a specific IP address or subnet. Regardless I agree its better to bind it to another port and have the user open/close that port based on their needs rather than having to restart the server every time to enable/disable web UI access.

Will it be possible to route the ACME CA challenge via port 80 for the admin domain and still display the admin domain on another port using the certificate generated via port 80? It should be as easy as allowing "/.well-known/acme-challenge" for the admin domain on port 80 but blocking all web UI routes unless it originates from the web UI port?

**Comment by anderspitman** [comment]
Actually this is probably going to be a bit trickier. The admin UI uses the same port as the tunnels, and uses SNI to route traffic. That means this needs to be done at the TLS level, not the port level. I'm currently working on a new tunneling protocol, which may affect how this is implementing, so I might wait on that before deciding on it.

**Comment by WGrobler** [comment]
You are right @anderspitman, I forgot that the ports are not passed to http.HandleFunc. This will complicate the different port solution. I agree, lets wait on this, other changes in the pipeline can influence implementation of this. 

@labeldevops, if you are interested in how this will work, I've added this with [commit 2e0ff06](https://github.com/WGrobler/boringproxy/commit/b496a8f9390975bdced214fc4048c4e75236e170). I don't think this will be merged into master, but you can use it if you want

**Comment by rfenouil** [comment]
Hi, I would like to +1 this request, I'd rather not expose/run the admin gui to the web.
I also think the implementation is limiting for people using free ddns services with limited number of subdomains and no wildcard.

An option to run the admin interface on a separate port would work for me (would it be a problem for generating certificate ? Maybe I'm wrong but I don't think I need https for this connection actually...). 
On top of that it would be even better to make it a separate process that I can start when needed only.

PS: BTW, thank you for the amazing tool, impressive piece of work :)
